src/components/dashboard-navbar.js

```diff
@@ -103,55 +121,109 @@ export const DashboardNavbar = (props) => {
-      .catch((error) => {
-        console.error(error)
-      });
-      console.log(new Date().getTime())
-      const newUser = {
-        wallets: [
-          {
-            name: "Default Account",
-            blockchain: "Ethereum", // TODO: enable other login accounts, for now only Ethereum account
-            chainId: network.chainId, 
-            address: address,
-            createdAt: new Date().getTime()
-          }
-        ],
-        signature: signature,
-        signerAddress: address
-      };
-      checkAndCreateUser(newUser).catch((error) => {
-        console.error(error)
-      })
-      dispatch(userActions.userAccount(address));
-      dispatch(userActions.connecion());
-      localStorage.setItem("userAddress", address);
+  const handleClickConnectWalletMenu = (event) => {
+    connectWalletSetAnchorEl(event.currentTarget);
+  };
+  const handleCloseConnectWalletMenu = () => {
+    connectWalletSetAnchorEl(null);
+  };
+
+  const connectEthereumWallet = async () => {
+    {
+      let provider;
+      let library;
+      let accounts;
+      let network;
+      try {
+        provider = await getWalletProvider().connect();
+        library = new ethers.providers.Web3Provider(provider);
+        accounts = await library.listAccounts();
+        network = await library.getNetwork();
+      } catch (error) {
+        console.error(error);
+      }
+      setProvider(provider);
+      setLibrary(library);
+      setNetwork(network);
+      setSelectedBlockchain("Ethereum")
+
+      console.log(network);
+
+      if (accounts) {
+        const address = accounts[0];
+        const signature = await signMessage(library)
+          .catch((error) => {
+            console.error(error)
+          });
+        console.log(new Date().getTime())
+        const newUser = {
+          wallets: [
+            {
+              name: "Default Account",
+              blockchain: selectedBlockchain, // TODO: enable other login accounts, for now only Ethereum account
+              chainId: network.chainId,
+              address: address,
+              createdAt: new Date().getTime()
+            }
+          ],
+          signature: signature,
+          signerAddress: address
+        };
+        checkAndCreateUser(newUser).catch((error) => {
+          console.error(error)
+        })
+        dispatch(userActions.userAccount(address));
+        dispatch(userActions.connecion());
+        dispatch(userActions.blockchain("Ethereum"));
+        localStorage.setItem("userAddress", address);
+        localStorage.setItem("blockchain", "Ethereum");
+      }
+    };
+  }
+
+  const connectSolanaWallet = async () => {
+    if (connected) {
+      disconnectSolWallet()
     }
-  }; 
+    setSelectedBlockchain("Solana")
+    setOpenSolanaWalletDialog(true)
+  }
+  const completeSolanaWalletConnect = async () => {
+    // console.log(publicKey.toBase58());
+    //sign 
+    const message = ["Log in CrossPay", userAddress];
```

Comment:
Why do we need the user address here?

Reply:
![image](https://user-images.githubusercontent.com/39519012/215118441-a3d36bf5-7a10-491c-92f0-d95659dcbbba.png)

Reply:
just following the same pattern as the eth wallet sign in... but i did make a mistake here, it should be the phantom wallet public key